trappy/ftrace.py

```diff
@@ -306,14 +360,18 @@ def __populate_data(self, fin, cls_for_unique_word):
 
             if not self.basetime:
                 self.basetime = timestamp
-
-            if (timestamp < self.window[0] + self.basetime) or \
-               (timestamp < self.abs_window[0]):
+                # Disregard windows if updating/creating cache
+                # This way the cache is ready for later window requests
+                if not self.__class__.disable_cache:
+                    self.max_window = self._max_window()
```

Comment:
IIUC this is set in two separate places depending on whether we're reading from the cache, but it's actually independent of that. Can we just do it in one place at the beginning? Or even make `self.max_window` a `property`?

Reply:
That's what I wanted to do, sadly it **needs** to be set in two separate places. If we have the cache, we can read the `basetime` parameter from it and create the max window. If not, we must start parsing the file to finally get the `basetime` and then we can create that max window value.